### PR TITLE
Use `--cfg curve25519_dalek_backend` to select backend

### PR DESCRIPTION
**Pull request by tarcieri**
Crate features are intended to be additive, whereas only 1-of-N possible backends can be selected.

Features can also be activated by transitive dependencies, which leads to a problem of different dependences selecting conflicting backends. Using `--cfg` instead moves all backend selection control to the toplevel executable.

This commit switches to the following RUSTFLAGS to enable backends:

- `--cfg curve25519_dalek_backend="fiat"`: uses `fiat-crypto` backend
- `--cfg curve25519_dalek_backend="simd"`: uses nightly-only SIMD backend

### CODE REVIEW

File: src/backend/mod.rs [hunk 1]
```diff
@@ -36,5 +36,5 @@
 
 pub mod serial;
 
-#[cfg(any(feature = "simd_backend", docsrs))]
+#[cfg(any(dalek_backend = "simd", docsrs))]
```
[the remark this answers is not in the document]
Comment (tarcieri) [reply src/backend/mod.rs:39]: Alternative suggestions?

[the remark this answers is not in the document]
Comment (jrose-signal) [reply src/backend/mod.rs:39]: The boring obvious one is `curve25519_dalek_backend`, if no one has anything better.

[the remark this answers is not in the document]
Comment (tarcieri) [reply src/backend/mod.rs:39]: A little long but definitely clear

[the remark this answers is not in the document]
Comment (tarcieri) [reply src/backend/mod.rs]: Changed to `curve25519_dalek_backend`

**Comment by tarcieri** [comment]
This is code complete.

It still needs corresponding documentation changes.

**Review by rozbb — APPROVED** [review]
I think this is good to go. The new backend selection docs can be in a new PR, once #453 lands.

**Review by pinkforest — APPROVED** [review]
Looks great :)

**Comment by jrose-signal** [comment]
I'm sorry I didn't look at this sooner; why is `bits` a separate axis?

**Comment by jrose-signal** [comment]
Yes, but future backends may not come in such pairs, and the `bits` selection doesn't necessarily correspond to your target architecture (as we've discovered), and it's diverging further from the 3.x branch.

**Comment by pinkforest** [comment]
My initial PR was embedding it with backend and it became horrondeous to maintain not to mention merge conflicts around where we needed to put cfg() gates between as there is cross-use.

**Comment by pinkforest** [comment]
If future backend doesn't need the selection then people don't need to set those bits - this is just an override not a default.
Default is selected automatically and then there is a further lotto over that: 
- https://github.com/dalek-cryptography/curve25519-dalek/pull/465
I'm documenting it here: 
- https://github.com/dalek-cryptography/curve25519-dalek/pull/467

**Comment by tarcieri** [comment]
Yeah, doing anything complex with `cfg` is something of a nightmare.

@jrose-signal I suggest you review #465. It should hopefully eliminate the need for manual `curve25519_dalek_bits` configuration for your use cases

**Pull request merged**